### PR TITLE
fix: `FilterGenesRegression` deprecated transpose and missed args; `ScTransformR` env config

### PR DESCRIPTION
**Pull request by xingzhongyu**
Fixed some errors in FilterGenesRegression and configuration environment issues in ScTransformR
Try to clear useless git information

### CODE REVIEW

File: dance/transforms/filter.py [hunk 1]
```diff
@@ -643,30 +643,30 @@ def __call__(self, data):
         func_dict = {"enclasc": self._filter_enclasc, "seurat3": self._filter_seurat3, "scmap": self._filter_scmap}
         if (filter_func := func_dict.get(self.method)) is None:
             raise ValueError(f"Unknown method {self.method}, supported options are: {list(func_dict)}.")
-
         data.data.obsm[self.out] = filter_func(feat)
         return data
 
-    def _filter_enclasc(feat: np.ndarray, num_genes: int, logger: Logger, no_check: bool = False) -> np.ndarray:
+    def _filter_enclasc(self, feat: np.ndarray, num_genes: int = 2000, logger: Logger = default_logger,
```
Comment (xingzhongyu) [inline dance/transforms/filter.py:649]: add self,num_genes,logger

File: dance/transforms/filter.py [hunk 2]
```diff
@@ -643,30 +643,30 @@ def __call__(self, data):
...
+                        no_check: bool = False) -> np.ndarray:
         logger.info("Start generating cell features using EnClaSC")
-
         num_feat = feat.shape[1]
         scores = np.zeros(num_feat) - 100
 
         feat_mean = feat.mean(0)
         drop_feat = (feat == 0).mean(0)
-        select_index = 0 < drop_feat < 1
+        select_index = (0 < drop_feat) & (drop_feat < 1)
```
Comment (xingzhongyu) [inline dance/transforms/filter.py:657]: It is an array,  need to use this form

File: dance/transforms/filter.py [hunk 3]
```diff
@@ -643,30 +643,30 @@ def __call__(self, data):
...
 
         x1 = feat_mean[select_index].reshape(-1, 1)
-        x2 = feat_mean[select_index].reshape(-1, 1)
+        x2 = drop_feat[select_index].reshape(-1, 1)
```
Comment (xingzhongyu) [inline dance/transforms/filter.py:660]: In my understanding, it should be like this

File: dance/transforms/filter.py [hunk 4]
```diff
@@ -643,30 +643,30 @@ def __call__(self, data):
...
         y = np.log(feat_mean + 1)[select_index].reshape(-1, 1)
 
-        y_pred = LinearRegression(x2, y).predict(x2)
+        y_pred = LinearRegression().fit(x2, y).predict(x2)
         scores[select_index] = (2 * y - y_pred - x1).ravel()
         feat_index = np.argpartition(scores, -num_genes)[-num_genes:]
-        return feat[:, feat_index]
+        return feat[feat_index, :].T
```
Comment (xingzhongyu) [inline dance/transforms/filter.py:666]: The array has been transposed before

File: dance/transforms/normalize.py [hunk 5]
```diff
@@ -114,43 +115,52 @@ class ScTransformR(BaseTransform):
...
-
-        if isinstance(data.data.X, sp.spmatrix):
-            self.logger.warning("Native support for sparse matrix is not implemented yet, "
-                                "converting to dense array explicitly.")
-            data.data.X = data.data.X.A
-        adata = data.X.copy()
+        with localconverter(robjects.default_converter):
+            utils = rpackages.importr('utils')
+            if self.mirror_index != -1:
+                utils.chooseCRANmirror(ind=self.mirror_index)
```
Comment (xingzhongyu) [inline dance/transforms/normalize.py:132]: The R package needs to be installed

File: dance/transforms/normalize.py [hunk 6]
```diff
@@ -114,43 +115,52 @@ class ScTransformR(BaseTransform):
...
+            seurat@assays$RNA<-seurat@assays$originalexp
+            seurat_p=SCTransform(seurat, vst.flavor = "v2", verbose = FALSE,min_cells={self.min_cells})
+            '''
+            r(r_code)
+            r_floatmatrix = r('seurat@assays$RNA@data')
+
+            # Convert to anndata
+            adata.X = r_floatmatrix.T
+            data.data.X = adata.X
+            return data
```
Comment (xingzhongyu) [inline dance/transforms/normalize.py:163]: It is best to keep it in the same environment with sce

**Review by RemyLau — APPROVED** [review]

**Pull request merged**